PsfShimMonitor/KernelTrace.cs

```diff
@@ -0,0 +1,1356 @@
+// Copyright (C) TMurgent Technologies. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+//
+// NOTE: PsfMonitor is a "procmon"-like display of events captured via the PSF TraceShim.
+
+using Microsoft.Diagnostics.Tracing;  // consumer
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session; // controller
+using System;
+using System.ComponentModel;  // backgroundworker
+using System.Threading;
+using System.Windows;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;  // ObservableCollection
+
+namespace PsfMonitor
+{
+    public partial class MainWindow : Window
+    {
+        public TraceEventSession myTraceEventSession_ProcsKernel = null;
+        public bool rememberToDisableSession_ProcsKernel = false;
+        public bool PleaseStopCollecting = false;
+        public int FilterOnProcessId = -1;
+
+        public List<EventItem> _TKernelEventListItems = new List<EventItem>();
+        public Object _TKernelEventListsLock = new Object();
+
+        Dictionary<UInt64, string> _KCBs = new Dictionary<UInt64, string>();
+        Dictionary<UInt64, string> _TKCBs = new Dictionary<UInt64, string>();
+        public Object _TKCBsListLock = new object();
+
+        private int MAX_KCBS = 100000;
+
+        private void KernelTraceInBackground_Start()
+        {
+            if (kerneleventbgw == null)
+                kerneleventbgw = new BackgroundWorker();
+            else if (kerneleventbgw.IsBusy)
+                return;
+            else
+                kerneleventbgw = new BackgroundWorker();
+
+
+
+            // Do processing in the background
+            kerneleventbgw.WorkerSupportsCancellation = true;
+            kerneleventbgw.WorkerReportsProgress = true;
+            kerneleventbgw.DoWork += KernelTrace_DoWork;
+            kerneleventbgw.ProgressChanged += KernelTrace_ProgressChanged;
+            kerneleventbgw.RunWorkerCompleted += KernelTrace_RunWorkerCompleted;
+            kerneleventbgw.RunWorkerAsync();
+        } // ETWTraceInBackground_Start()
+
+        private void KernelTrace_DoWork(object sender, DoWorkEventArgs e)
+        {
+            BackgroundWorker worker = sender as BackgroundWorker;
+            Thread.CurrentThread.Name = "ETWReader";
+
+            EnableKernelTrace(worker);
+        }
+
+        private void KernelTrace_ProgressChanged(object sender, ProgressChangedEventArgs e)
+        {
+            try
+            {
+                lock (_TKernelEventListsLock) //_TKCBsListLock)
+                {
+                    if (_TKCBs.Count > 0)
+                    {
+                        foreach (UInt64 k in _TKCBs.Keys)
+                        {
+                            // avoid catches
+                            string s = null;
```

Comment:
Please do not use single letter variable names.

Reply:
updated.